crates/bevy_ecs/src/schedule/state.rs

```diff
@@ -579,19 +573,50 @@ pub trait StateSet: sealed::StateSetSealed {
     );
 }
 
-impl<S: States> StateSetSealed for S {}
+trait InnerStateSet: Sized {
```

Comment:
Doc comments on what this is and why it's useful please :) Very easy to get confused about this design in the future.

Reply:
Done